chips/nrf5x/src/rtc.rs

```diff
@@ -135,11 +135,11 @@ impl Time for Rtc<'a> {
     type Frequency = Freq32KHz;
 
     fn now(&self) -> u32 {
-        self.registers.counter.get()
+        self.registers.counter.read(Counter::VALUE)
     }
 
     fn max_tics(&self) -> u32 {
-        core::u32::MAX
+        (1 << 24) - 1
```

Comment:
this could be a constant instead

Reply:
I don't think it matters much, the compiler will optimize that to the constant anyway (and likely inline this function as well). Unless you think there should be a `u24::MAX` constant defined for the whole kernel?

Reply:
I think you are right w.r.t optimizations but I was more concerned with readability for this

Reply:
I think it would make sense to define the constant in the `Time` HIL as part of #1413.

Reply:
yeah, let's merge this and revisit this converation there :)